Feature/1678 edges bs3

richard-jones opened the pull request:
This code has now been tested by the DOAJ team and is ready for merge pending your confirmation that the code is fit for deployment.

Steven-Eardley commented:
TODO: check whether moving JS has removed optimisations to serving static content directly by nginx.

Steven-Eardley commented on portality/app.py:
```diff
@@ -145,32 +145,6 @@
-# alongside the DOAJ codebase.  I know they could also go into the
-# nginx config, but there is a chance that they will get lost or forgotten
-# some day, whereas this approach doesn't have that risk.
-@app.route("/doaj")
```
I do support removing the legacy routes, but are you aware of how frequently they're used? (~200k in last 3 weeks)

https://doajkibana.cottagelabs.com/app/kibana#/discover?_a=(columns:!(context.request.url.full),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:'apm-*',key:context.request.url.pathname,negate:!f,params:(query:%2Fdoaj,type:phrase),type:phrase,value:%2Fdoaj),query:(match:(context.request.url.pathname:(query:%2Fdoaj,type:phrase))))),index:'apm-*',interval:auto,query:(language:lucene,query:''),sort:!('@timestamp',desc))&_g=(refreshInterval:(pause:!t,value:0),time:(from:now-30d,mode:quick,to:now))

Steven-Eardley replied:
Although you may notice some odd functions requested, like 'abstract'

richard-jones replied:
Ok, I think we should discuss this in the team meeting, and decide what to do.  I think maintaining them for 5 years is probably enough, but perhaps we can do something softer than just turn them off abruptly.  Or maybe this update is just not the right update in which to make that call, in which case we can revert the change.

richard-jones replied:
In fact, I'm just going to do that, and open an issue for removing legacy url support.

Steven-Eardley commented:
This really is too big to review. If you have any specific files I should look at, give me links. Apart from that, having the tests pass and its stability on the test server will have to suffice.

Steven-Eardley commented:
4 test failures when run locally on my machine:

```
Failure
Traceback (most recent call last):
  File "/usr/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File "/home/steve/code/cl/doaj/doajtest/unit/test_fc_editor_app_review.py", line 103, in test_01_editor_review_success
    assert re.search(eg_rx, html)
AssertionError: 

Failure
Traceback (most recent call last):
  File "/usr/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File "/home/steve/code/cl/doaj/doajtest/unit/test_fc_editor_journal_review.py", line 93, in test_01_editor_review_success
    assert re.search(eg_rx, html)
AssertionError: 

Failure
Traceback (most recent call last):
  File "/usr/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File "/home/steve/code/cl/doaj/doajtest/unit/test_fc_publisher_update_request.py", line 67, in test_01_publisher_update_request_success
    assert re.search(pissn_rx, html)
AssertionError: 

Failure
Traceback (most recent call last):
  File "/usr/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File "/home/steve/code/cl/doaj/doajtest/unit/test_formrender.py", line 119, in test_02_subclass_render_field_group
    assert "disabled" in html
AssertionError
```

richard-jones commented:
tests are now fixed, and all passing locally for me.  Just waiting to see if they all make it through codeship

richard-jones commented:
It's difficult to point you at a specific file which is more important than any other.  I guess if you want to check overall code style/approach, a sample of important ones would be:

* The public search interface: https://github.com/DOAJ/doaj/blob/feature/1678_edges_bs3/portality/static/js/edges/public.journalarticle.edge.js
* An admin search interface: https://github.com/DOAJ/doaj/blob/feature/1678_edges_bs3/portality/static/js/edges/admin.journalarticle.edge.js
* A template with a search interface embedded: https://github.com/DOAJ/doaj/blob/feature/1678_edges_bs3/portality/templates/doaj/search.html
* A complicated form page, with all the new section imports: https://github.com/DOAJ/doaj/blob/feature/1678_edges_bs3/portality/templates/formcontext/maned_application_review.html

If you like the look of those pages, then 80% of everything else that was done would be to your liking.

The remaining 20% is distributed finely throughout the codebase in the form of some directory reorganisations, base template edits, cleaning up web routes in views, etc.  None of it particularly exciting or dramatic, so provided that the application actually works they should all be fine.

richard-jones commented on portality/static/js/doaj.js:
```diff
@@ -0,0 +1,126 @@
+            alert("We weren't able to set your cookie consent preferences, please try again later.");
+        }
+    })
+}
```
I think for the moment I'm not going to do this.  The reason is that everything else on the site is ES5 compatible, and doing this would make this one file ES6. I've so far held off on moving to ES6 features, because of the various old browsers that we want to support.  I'm not against looking to upgrade, but I don't think this PR is the place to do it.

richard-jones commented on portality/static/js/multiformbox.js:
```diff
@@ -0,0 +1,345 @@
+
+            this.confirmSubmit = function (data) {
+                if (data.error) {
+                    alert("There was a problem processing your request.  " + data.error);
```
HTML compacts multiple spaces to one, so I don't think this needs to change.  Also, two spaces after a full stop is hard-wired into my brain, so you'll find this everywhere.  But it doesn't matter from a web perspective.

Steven-Eardley commented:
This is looking fantastic, I just have a few more oddities from gazing upon it in my browsers:

<img width="429" alt="safari_frontpagesearch" src="https://user-images.githubusercontent.com/1446355/65591193-ee1b7f00-df83-11e9-8b4d-43ace4e305b0.png">

The front page search looks good in safari (above) but firefox (below) has a missing border around the search button

![ffx_frontpage_search](https://user-images.githubusercontent.com/1446355/65591254-07bcc680-df84-11e9-9647-842e80fd1c66.png)

Next, do we want to make the search button the same colour on the search page, or is changing from orange to blue okay? (the blue looks great, but it's not consistent) I guess this is more of a design thing, so perhaps not worth thinking about now.

Next, there's an odd missing space inthe clear search button in safari when simulating an iPad in portrait mode:

<img width="643" alt="ipad_portrait" src="https://user-images.githubusercontent.com/1446355/65591359-494d7180-df84-11e9-9bc9-3a79a4f26722.png">

the equivalent settings in firefox don't have an issue:

![ffx_ipad](https://user-images.githubusercontent.com/1446355/65591396-5b2f1480-df84-11e9-872c-be16e3192d37.png)

richard-jones commented:
I have made the following fixes:

* Changed the front page search button to match the search interface button.  This may well get changed when styled, but for the moment it was behaving weirdly in firefox as it was, and styling it the same makes it work properly

* Added a space to the "Clear search filters" prefix in responsive view on search.  I can't verify that it works, but it should, and if it doesn't we can always hotfix it later.

I'm now going to go ahead and set a date for release, which I suggest is Tuesday after our meeting next week.

richard-jones commented:
In the mean time, you may now wish to synchronise your Python 3 branch
